Ash Drake Bug Fix and Latency Accounting

Whoneedspacee opened the pull request:
:cl: Whoneedspacee
balance: ash drake fire does less damage now
balance: ash drake takes longer to swoop down now
balance: tiles take longer to fully convert into lava now, slowing down the arena attack as well
balance: fire breath now moves slower
balance: triple fire breath for the lava swoop only happens below half health now
fix: https://github.com/tgstation/tgstation/issues/40621#issue-365415326
/:cl:

[why]: Complaints of ash drake being too strong in the lag have certainly not gone unheard, as well as the insults. The mech bug has been fixed from https://github.com/tgstation/tgstation/issues/40621#issue-365415326 as well as making ash drake a lot slower and less punishing.

Whoneedspacee committed:
Drake meteors are replaced with lava spawning out of the ground, deals damage on contact as well as normal lava damage, turns back after a period of time.

Fixes a bug where none of ash drakes attacks could damage mechs.

Adds directional fire to ash drake

Ash drake swoops no longer fly towards you but teleport above you, this was a non-threatening boring attack previously which is why it has been removed.

Slightly delays the time for meteors to hit the ground.

Adds cooldowns for the swoop attacks, previously there could be barely any time before ash drake would once again swoop up, unable to be attacked.

Whoneedspacee committed:
Ash drake now shoots fire when he cannot swoop.

Whoneedspacee committed:
Removes drakes ability to shoot fire after swoops, decreases swooping cooldown for triple swooping

Whoneedspacee committed:
Removes useless code and adds footsteps.

Whoneedspacee committed:
Adds the old drake fireball into the admin punishments file so it can still be used

Removes the old mass lava changing attack, replaced with placing lava around you and swooping on top of that

Removes useless async calls

Adds a new wall to be used for the arena attack

Changes the name of the target visual to lava_warning for more clarity

Slows down the fire on the ash drake fire cone attack

There is now a spot for all mobs with clients in the arena attack

Whoneedspacee committed:
Removes useless swoop cooldown variable and uses parent megafauna mob recovery_time variable

Adds new lava swoop attack and a triplet variant to replace the old triple swooping attack

Replaces ash drake message warnings with src mob name for clarity

Fire line attack no longer includes source turf

Fixes a bug where the fire line attack would damage the ash drake himself

Fixes a bug where the ash drakes fire breath could hit you multiple times

Increases the speed of ash drakes fire attack

Alphas out the lava creation attacks more at the start of the animation for more discrepancy from already existing lava

Slightly increases the damage of direct lava hits to players

Whoneedspacee committed:
Changes turf in arena to basalt instead of being dumb and spawning the empty spaces in walls

Adds a waiting period in the lava arena attack so there's time to realize what attack is actually happening

Adds a new safe tile which makes it more apparent to the player where they need to go

Fixes a bug where the ash drake would shoot fire after dying in the middle of the lava swoop attack

Fixes a bug where the fire line attack could try to calculate target positioning without a target

Decreases fire breath direct damage

Re-adds swooping animation to fire drake but faster

Decreases direct lava damage again

Whoneedspacee committed:
Slightly decreases mech direct damage

Removes useless swoop_speed variable

Decreases descent time variable

Removes useless goto_target() function

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into drake

Whoneedspacee committed:
Changes SetRecoveryTime method to work properly by changing the ranged cooldown, this makes it so setrecoverytime won't require manual checks in the openfire function, and will instead halt all instances of it being called for the time. This only affects the drake file as it's the only one that SetRecoveryTime is used in, but will aid in future megafauna alterations and cuts down on a check in the drake file.

Slightly increases the fire lines damage to be more threatening, it was face tankable previously.

Whoneedspacee committed:
Fixes bug where mechs wouldn't have safe spots spawned for them.

Whoneedspacee committed:
Triple fire breath only happens below half health now

Decreases fire breath damage

Increases time until tiles turn into lava

Slows down fire breath

Mechs now have a space in the lava arena

Dragon swoop going down takes twice as long as original drake

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into drake

# Conflicts:
#	code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm

Whoneedspacee committed:
Fixes descent time

Mechs now take damage from dragons swooping down and landing on them

ExcessiveUseOfCobblestone commented:
`fixes #40621` needed in description

Borisvanmemes commented:
Thank you deeply for this.

Repukan commented:
Don't give an inch to people that want free loot, they'll take the whole yard

Whoneedspacee commented:
I’m not going to edit the other megafauna at this point since I want to work on my station defense gamemode. I’m also unable to test these in the lag due to being banned, which makes it harder to edit megafauna and have them be balanced.

It’s still harder than it originally was and is more fun, so making it just a little bit easier isn’t really a big deal to me.

Either way the lack of support towards the update didn’t make it much easier to keep it as it was, and I just want to be done with something that people continue to shit on.

ExcessiveUseOfCobblestone commented:
when you touch powerful items there will always be a crowd that cries.

XDTM commented:
I'm also kinda against the kind of player who is experienced enough to still beat hardmode bosses and wants to pull the ladder up so the loot is their exclusive

Iamgoofball commented:
@Whoneedspacee you should bug admins to see if you can get a pass for testing

subject217 commented:
If Whoneedspacee wants to be unbanned he should appeal it.

Repukan commented:
If the people bitching about end game loot being hard to get want it nerfed, they should do it themselves. The fact that none of them have put up a PR means they don't care enough about balance and only care about their OP loot not being brain dead to obtain

Daxxed commented:
> 
> 
> If the people bitching about end game loot being hard to get want it nerfed, they should do it themselves. The fact that none of them have put up a PR means they don't care enough about balance and only care about their OP loot not being brain dead to obtain

Some of us may not know it's up to themselves to fix other's garbage updates.
Dude is even banned from the server, if this isn't "I can't kill dragons so now you can't either" I don't know.

Anyway @Whoneedspacee thank you for hearing our prayers and reconsidering.

Whoneedspacee commented:
> > If the people bitching about end game loot being hard to get want it nerfed, they should do it themselves. The fact that none of them have put up a PR means they don't care enough about balance and only care about their OP loot not being brain dead to obtain
> 
> Some of us may not know it's up to themselves to fix other's garbage updates.
> Dude is even banned from the server, if this isn't "I can't kill dragons so now you can't either" I don't know.
> 
> Anyway @Whoneedspacee thank you for hearing our prayers and reconsidering.

I'm almost a little insulted that I loved megafauna content and wanted to make something more interesting, and everyone shits on me for taking a boring boss and trying to make it interesting. Just because I'm banned doesn't mean I hate the server in any fucking way.

I don't care that people powergame in ss13 mining because there's easier ways to do it to be honest. What I care about is me taking 12 hours a day out of my week spending constant time trying to think of attacks that fit the boss and work with ss13's limitations in programming, and people trash talk me for not making it perfect on the first go, and don't give me clear help on how I can fix it, or even record a simple couple minute video to help me see from their perspective.

I'm not nerfing it because I think it's a hard boss, even in lag, I'm nerfing it because I don't want to hear people complaining about it anymore.

Daxxed commented:
We DO appreciate the effort but I don't think anyone asked for this rework.
Add a new boss if you wish, make it the maddest hardest thing to ever roam lavaland and lock the fireball, dragon's blood, spellblade and hierophant's club behind it I don't care.
But please don't take away my joy of killing dragons and the drake suit ( sitting afk in ash storms is not fun gameplay )
Also you have to keep in mind the server plays alot differently than a localhost, even with ping as low as 15, not to mention you always spawn the dragon in the open field, have you tried aggroing it in a small corridor and dodging the initial swoop which it almost always opens with ?

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm:
```diff
@@ -269,7 +272,7 @@ Difficulty: Medium
 	
 	// Ash drake flies onto its target and rains fire down upon them
-	var/descentTime = 7;
+	var/descentTime = 10;
```
this probably wasn't a problem given that it was at 0.5 before

Whoneedspacee replied:
This is increasing the time it takes for the dragon to hit the ground once he flies above you, what are you talking about?

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm:
```diff
@@ -137,10 +137,11 @@ Difficulty: Medium
-	fire_cone()
-	sleep(10)
-	fire_cone()
+	if(health < maxHealth/2)
```
`*0.5` rather than `/2`

Whoneedspacee replied:
Why?

iskyp replied:
multiplication is preferred over division for ???

Whoneedspacee replied:
Should I edit the other occurences of this in the file then? I was just taking from what was there.

Whoneedspacee committed:
Division is now illegal

Whoneedspacee committed:
Lava pools now spawn closer and spawn slower near you

Lesser ash drake now has minor lava pools in his targeted swoop

Drakewall duration fixed to last to the end of the animations

optimumtact commented:
stop making changes you berrypicker
